src/irmin-pack/unix/async_intf.ml

```diff
@@ -0,0 +1,50 @@
+(*
+ * Copyright (c) 2022-2022 Tarides <contact@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+module type S = sig
+  (** Basic abstraction for a worker. *)
+
+  type task
+
+  type outcome = [ `Success | `Cancelled | `Failure of string ]
+  [@@deriving irmin]
+
+  type status = [ outcome | `Running ] [@@deriving irmin]
+
+  val async : (unit -> unit) -> task
+  (** Start a task. *)
+
+  val await : task -> [> outcome ] Lwt.t
+  (** If running, wait for a task to finish and return its outcome.
+
+      If not running, return the oucome of the task. *)
+
+  val status : task -> [> status ]
+  (** If running, refresh the status of the task, without blocking.
+
+      If not running, return the oucome of the task. *)
+
+  val cancel : task -> unit
+  (** If running, cancel the task.
+
+      If not running, do nothing. *)
+end
+
+module type Sigs = sig
+  module type S = S
+
+  module Unix : S
```

Comment:
I think it would be nicer to directly use `Async` instead of `Async.Unix` since we are already in a unix context.

Reply:
It's there to mimic `Io.Unix`, which is here to one day support mirage. Let's keep it that way if you agree